diff --git a/src/libudev.c b/src/libudev.c
index d954dae..d663160 100644
--- a/src/libudev.c
+++ b/src/libudev.c
@@ -229,7 +229,6 @@ UDEV_EXPORT struct udev *udev_new(void)
                         }
                         if (strcmp(key, "udev_rules") == 0) {
                                 set_value(&udev->rules_path[0], val);
-                                udev->rules_path_count = 1;
                                 continue;
                         }
                 }
@@ -259,18 +258,18 @@ UDEV_EXPORT struct udev *udev_new(void)
                 udev->rules_path[0] = strdup(PKGLIBEXECDIR "/rules.d");
                 if (!udev->rules_path[0])
                         goto err;
+        }
 
-                /* /run/udev -- runtime rules */
-                if (asprintf(&udev->rules_path[2], "%s/rules.d", udev->run_path) < 0)
-                        goto err;
+        /* /run/udev -- runtime rules */
+        if (asprintf(&udev->rules_path[2], "%s/rules.d", udev->run_path) < 0)
+                goto err;
 
-                /* /etc/udev -- local administration rules */
-                udev->rules_path[1] = strdup(SYSCONFDIR "/udev/rules.d");
-                if (!udev->rules_path[1])
-                        goto err;
+        /* /etc/udev -- local administration rules */
+        udev->rules_path[1] = strdup("/etc/udev/rules.d");
+        if (!udev->rules_path[1])
+                goto err;
 
-                udev->rules_path_count = 3;
-        }
+        udev->rules_path_count = 3;
 
         dbg(udev, "context %p created\n", udev);
         dbg(udev, "log_priority=%d\n", udev->log_priority);
